gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineFormat.java

```diff
@@ -41,6 +55,104 @@
+
+        SpotlessExtension spotlessExtension = project.getExtensions().getByType(SpotlessExtension.class);
+        // Keep spotless from eagerly configuring all other tasks.  We do the same thing as the enforceCheck
+        // property below by making the check task depend on spotlessCheck.
+        // See  https://github.com/diffplug/spotless/issues/444
+        spotlessExtension.setEnforceCheck(false);
+
+        // Allow disabling copyright for tests
+        if (!"false".equals(project.findProperty("com.palantir.baseline-format.copyright"))) {
+            configureCopyrightStep(project, spotlessExtension);
+        }
+
+        // necessary because SpotlessPlugin creates tasks in an afterEvaluate block
+        TaskProvider<Task> formatTask = project.getTasks().register("format", task -> {
+            task.setGroup("Formatting");
+        });
+        project.afterEvaluate(p -> {
+            formatTask.configure(t -> {
+                t.dependsOn("spotlessApply");
+            });
+
+            // re-enable spotless checking, but lazily so it doesn't eagerly configure everything else
+            project.getTasks().named(JavaBasePlugin.CHECK_TASK_NAME).configure(t -> {
+                t.dependsOn(project.getTasks().named("spotlessCheck"));
+            });
+
+            // The copyright step configures itself lazily to allow for baselineUpdateConfig to potentially create the
+            // right files. Therefore, also make sure that these will run in the right order.
+            project.getPluginManager().withPlugin("com.palantir.baseline-config", baselineConfig -> {
+                project.getTasks()
+                        .matching(t -> t.getName().startsWith("spotless"))
+                        .configureEach(t -> t.mustRunAfter("baselineUpdateConfig"));
+            });
+        });
+
+        project.getPluginManager().withPlugin("java", plugin -> {
+            configureSpotlessJava(project, spotlessExtension);
+        });
+    }
+
+    /**
+     * Necessary in order to not fail right away if the copyright folder doesn't exist yet, because it would be created
+     * by {@code baselineUpdateConfig}.
+     */
+    private FormatterStep createLazyLicenseHeaderStep(Project project) {
+        // Spotless will consider the license header to be the file prefix up to the first line starting with delimiter
+        String delimiter = "(?! \\*|/\\*| \\*/)";
+
+        // Compute the copyright lazily
+        Supplier<FormatterStep> realStep = Suppliers.memoize(() -> {
+            String header = computeCopyrightHeader(project);
+            return LicenseHeaderStep.createFromHeader(header, delimiter);
+        });
+
+        return (FormatterStep) Proxy.newProxyInstance(
+                Thread.currentThread().getContextClassLoader(),
+                new Class[] {FormatterStep.class},
+                (_proxy, method, args) -> method.invoke(realStep.get(), args));
```

Comment:
can we use a LazyInitializedFormatterStep containing a memozied suppleire instead pls?

Reply:
yep, and turns out doing so now showcased that this thing was trivially serializable which is not quite what you want.